todoapp/src/components/Todo.tsx

```diff
@@ -168,13 +184,83 @@ const useStyles = makeStyles({
         }
       }
     }
-  }
+  },
 })
 
+
+type Todo = {
+  id: string;
+  text: string;
+  completed: boolean
+}
+
 function Todos(): JSX.Element {
   const isSmallScreen = useMediaQuery('(max-width:600px)');
   const classes = useStyles();
+  const { todosState } = useGlobalState();
+  const [text, setText] = useState("");
+  const [selectedTodoState, setSelectedTodoState] = useState<string | null>(null);
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setText(e.target.value);
+  }
+  const createTodo = (e: FormEvent) => {
+    e.preventDefault();
+    if (selectedTodoState) {
+      todosState.set((prevTodos: Todo[]) => {
+        const updatedTodos = prevTodos.map((todo) => {
```

Comment:
can you extract prevTodos from global state and then evaluate and then set to todoState after that?
This is complex to understand for new dev.